packages/block-library/src/post-featured-image/edit.js

```diff
@@ -308,8 +308,7 @@ export default function PostFeaturedImageEdit( {
 				mediaLibraryButton={ ( { open } ) => {
 					return (
 						<Button
-							// TODO: Switch to `true` (40px size) if possible
-							__next40pxDefaultSize={ false }
+							__next40pxDefaultSize
```

Comment:
This can be seen, while you add `core/post-featured-image` block.

**Note:** This has some styles that have higher specificity and specific to this upload button on this block, hence before and after is same, but we can see our change is applied in inspector controls.

| Before | After     |
| :-------- | :------- |
| <img width="1355" alt="featured-image-before" src="https://github.com/user-attachments/assets/8bb7fb2d-1eba-42fa-b688-7512a5d231e1"> | <img width="1355" alt="featured-image-after" src="https://github.com/user-attachments/assets/46cc927a-7019-449b-8ad2-bf15676060d1"> |

Reply:
As @hbhalodia said, noting that the `__next40pxDefaultSize` prop here doesn't have an impact because of the [style overrides](https://github.com/WordPress/gutenberg/blob/ff01bcbd4aeb84d6a28a655ca652a9ad9f16cebe/packages/block-library/src/post-featured-image/editor.scss#L10-L45) applied to the placeholder and the button components in the post featured image block.

These style overrides should be removed, if possible — but it's not in the scope of this PR. We can keep changes as they are.